sympy/solvers/solveset.py

```diff
@@ -1434,8 +1434,14 @@ def _solve_exponential(lhs, rhs, symbol, domain):
     a_term = a.as_independent(symbol)[1]
     b_term = b.as_independent(symbol)[1]
 
-    a_base, a_exp = a_term.base, a_term.exp
-    b_base, b_exp = b_term.base, b_term.exp
+    if a_term.is_Pow:
+        a_base, a_exp = a_term.base, a_term.exp
```

Comment:
Would it work to use
```
a_base, a_exp = a.as_base_exp()
```

Reply:
I think this is more suitable
```
    if a_term.is_Pow:
        a_base, a_exp = a_term.base, a_term.exp
    else: # it's just for the sake of domain
        a_base, a_exp = a.as_base_exp()
   
    if b_term.is_Pow:
        b_base, b_exp = b_term.base, b_term.exp
    else:
        b_base , b_exp = b.as_base_exp()
```

than to the present ....
should I proceed for another commit ?

Reply:
Do you need the `if`?

Reply:
Yes , because if a_term is `exp(-3)` the scenario could be like a_base < 0 (used in Line 1444 or 1450)
so it's necessary to consider separate case for `pow` terms hence if to be used

Reply:
Both branches give the same result in that case though:
```
In [13]: a_term = exp(-3)                                                                                

In [14]: a_term.base, a_term.exp                                                                         
Out[14]: (ℯ, -3)

In [15]: a_term.as_base_exp()                                                                            
Out[15]: (ℯ, -3)
```